test/click.jest.js

```diff
@@ -307,6 +307,177 @@ describe('Page.click', function() {
+      page.on('framenavigated', (frame) => {
+        if (frame.url().endsWith('/login-with-x-frame-options-deny.html')) {
+          fulfull(frame.url());
+        }
+      });
+    });
+
+    const consoleMessaged = new Promise(fulfill => {
+      page.on('console', msg => {
+        fulfill(msg.text());
+      });
+    });
+
+    const frame = page.frames()[1];
+    const button = await frame.$('#pt-login');
+    await button.click();
+
+    if (FFOX) {
+      expect(await navigated).toBeTruthy();
+    } else if (WEBKIT || CHROMIUM && HEADLESS) {
+      expect((await consoleMessaged).match(/^Refused to display.*login-with-x-frame-options-deny\.html' in a frame because it set 'X-Frame-Options' to 'DENY'\./i)).toBeTruthy();
+    } else if (CHROMIUM && !HEADLESS) {
+      expect((await consoleMessaged).match(/^Failed to load resource: the server responded with a status of 404 \(Not Found\)/i)).toBeTruthy();
+    }
+  })
+  it('should click and navigate to a x-frame-options:DENY link', async({page, server}) => {
+    server.setRoute('/login-with-x-frame-options-deny.html', async (req, res) => {
+      res.setHeader('Content-Type', 'text/html');
+      res.setHeader('X-Frame-Options', 'DENY');
+      res.end();
+    });
+
+    server.setRoute('/wikipedia.html', async(req, res) => {
+      res.setHeader('Content-Type', 'text/html');
+      res.end(`
+      <!DOCTYPE html>
+      <html>
+        <body>
+          <a id="pt-login" href="/login-with-x-frame-options-deny.html">login</a>
+        </body></html>
+      `)
+    })
+
+    server.setRoute('/wrapper.html', async(req, res) => {
+      res.setHeader('Content-Type', 'text/html');
+      res.end(`
+      <!DOCTYPE html>
+      <html>
+      <body>
+          <div>
+              <iframe src="${server.CROSS_PROCESS_PREFIX + '/wikipedia.html'}"  width="100%" height="100%" > </iframe>
+          </div>
+      </body></html>
+      `)
+    })
+
+    await page.goto(server.PREFIX + '/wrapper.html')
+    const navigated = new Promise(fulfull => {
+      page.on('framenavigated', (frame) => {
+        if (frame.url().endsWith('/login-with-x-frame-options-deny.html')) {
+          fulfull(frame.url());
+        }
+      });
+    });
+
+    const consoleMessaged = new Promise(fulfill => {
+      page.on('console', msg => {
+        fulfill(msg.text());
+      });
+    });
+
+    const frame = page.frames()[1];
+    const button = await frame.$('#pt-login');
+    await button.click();
+
+    if (FFOX) {
+      expect(await navigated).toBeTruthy();
+    } else if (WEBKIT || CHROMIUM && HEADLESS) {
+      expect((await consoleMessaged).match(/^Refused to display.*login-with-x-frame-options-deny\.html' in a frame because it set 'X-Frame-Options' to 'DENY'\./i)).toBeTruthy();
+    } else if (CHROMIUM && !HEADLESS) {
+      expect((await consoleMessaged).match(/^Failed to load resource: the server responded with a status of 404 \(Not Found\)/i)).toBeTruthy();
+    }
```

Comment:
@aslushnikov What should I baseline cases against? Consumer version of Firefox would fall into the  `Refused to display` category, but in Playwright it allows the frame navigation.